### PR TITLE
Add rules to check permissions and owner of important backup account files

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Add rules to check for owner, groupowner and permissions of:
- `/etc/passwd-`
- `/etc/shadow-`
- `/etc/group-`
- `/etc/gshadow-`

#### Notes:
- Mostly covers CIS requirements for RHEL7 and RHEL8. Sometimes the required permissions differed across the benchmarks, when it happened I went with the most restrictive.

### CODE REVIEW

**Comment by redhatrises** [comment]
> Mostly covers CIS requirements for RHEL7 and RHEL8. Sometimes the required permissions differed across the benchmarks, when it happened I went with the most restrictive.

Do the permissions conflict with what is specified in the RPM?

**Comment by yuumasato** [comment]
> Do the permissions conflict with what is specified in the RPM?

No RPM owns the backup files.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_group/rule.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Verify Group Who Owns Backup group File'
+
+description: '{{{ describe_file_group_owner(file="/etc/group-", group="root") }}}'
+
+rationale: |-
+    The <tt>/etc/group-</tt> file is a backup file of the <tt>/etc/group</tt>, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_group/rule.yml]: nitpick: `backup file of <tt>/etc/group</tt>, and as such, it also contains`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_gshadow/rule.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+documentation_complete: true
+
+title: 'Verify Group Who Owns Backup gshadow File'
+
+description: '{{{ describe_file_group_owner(file="/etc/gshadow-", group="root") }}}'
+
+rationale: |-
+    The <tt>/etc/gshadow-</tt> file is a backup of the <tt>/etc/gshadow</tt>, and as such it
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_gshadow/rule.yml]: nitpick: `backup file of <tt>/etc/gshadow</tt>, and as such, it contains`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_passwd/rule.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Verify Group Who Owns Backup passwd File'
+
+description: '{{{ describe_file_group_owner(file="/etc/passwd-", group="root") }}}'
+
+rationale: |-
+    The <tt>/etc/passwd-</tt> file is a backup file of the <tt>/etc/passwd</tt> file and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_passwd/rule.yml]: nitpick: `backup file of <tt>/etc/passwd</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_shadow/rule.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Verify User Who Owns Backup shadow File'
+
+description: '{{{ describe_file_owner(file="/etc/shadow-", owner="root") }}}'
+
+rationale: |-
+    The <tt>/etc/shadow-</tt> file is a backup file of the <tt>/etc/shadow</tt> file, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_shadow/rule.yml]: nitpick: `backup file of <tt>/etc/shadow</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_group/rule.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Verify User Who Owns Backup group File'
+
+description: '{{{ describe_file_owner(file="/etc/group-", owner="root") }}}'
+
+rationale: |-
+    The <tt>/etc/group-</tt> file is a backup file of the <tt>/etc/group</tt>, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_group/rule.yml]: nitpick: `backup file of <tt>/etc/group</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_gshadow/rule.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+documentation_complete: true
+
+title: 'Verify User Who Owns Backup gshadow File'
+
+description: '{{{ describe_file_owner(file="/etc/gshadow-", owner="root") }}}'
+
+rationale: |-
+    The <tt>/etc/gshadow-</tt> file is a backup of the <tt>/etc/gshadow</tt>, and as such it
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_gshadow/rule.yml]: nitpick: `backup file of <tt>/etc/gshadow</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_passwd/rule.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Verify User Who Owns Backup passwd File'
+
+description: '{{{ describe_file_owner(file="/etc/passwd-", owner="root") }}}'
+
+rationale: |-
+    The <tt>/etc/passwd-</tt> file is a backup file of the <tt>/etc/passwd</tt> file and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_passwd/rule.yml]: nitpick: `backup file of <tt>/etc/passwd</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+documentation_complete: true
+
+title: 'Verify Group Who Owns Backup shadow File'
+
+description: '{{{ describe_file_group_owner(file="/etc/shadow-", group="root") }}}'
+
+rationale: |-
+    The <tt>/etc/shadow-</tt> file is a backup file of the <tt>/etc/shadow</tt> file, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml]: nitpick: `backup file of <tt>/etc/shadow</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_group/rule.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
+documentation_complete: true
+
+title: 'Verify Permissions on Backup group File'
+
+description: |-
+    {{{ describe_file_permissions(file="/etc/group-", perms="0644") }}}
+
+rationale: |-
+    The <tt>/etc/group-</tt> file is a backup file of the <tt>/etc/group</tt>, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_group/rule.yml]: nitpick: `backup file of <tt>/etc/group</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_gshadow/rule.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
+documentation_complete: true
+
+title: 'Verify Permissions on Backup gshadow File'
+
+description: |-
+    {{{ describe_file_permissions(file="/etc/gshadow-", perms="0000") }}}
+
+rationale: |-
+    The <tt>/etc/gshadow-</tt> file is a backup of the <tt>/etc/gshadow</tt>, and as such it
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_gshadow/rule.yml]: nitpick: `backup file of <tt>/etc/gshadow</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_passwd/rule.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
+documentation_complete: true
+
+title: 'Verify Permissions on Backup passwd File'
+
+description: |-
+    {{{ describe_file_permissions(file="/etc/passwd-", perms="0600") }}}
+
+rationale: |-
+    The <tt>/etc/passwd-</tt> file is a backup file of the <tt>/etc/passwd</tt> file and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_passwd/rule.yml]: nitpick: `backup file of <tt>/etc/passwd</tt>, and as such, it`

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_shadow/rule.yml [hunk 12]
```diff
@@ -0,0 +1,39 @@
+documentation_complete: true
+
+title: 'Verify Permissions on Backup shadow File'
+
+description:  |-
+    {{{ describe_file_permissions(file="/etc/shadow-", perms="0000") }}}
+
+rationale: |-
+    The <tt>/etc/shadow-</tt> file is a backup file of the <tt>/etc/shadow</tt> file, and as such
```
Comment (redhatrises) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_shadow/rule.yml]: nitpick: `backup file of <tt>/etc/shadow</tt>, and as such, it`

**Comment by yuumasato** [comment]
Hopefully addressed all quirks in rationale language.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_group/rule.yml (hunk 1 above)
Comment (redhatrises) [reply linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_group/rule.yml]: @yuumasato missed the comma `,` after the words `and as such` in each of these.

Comment (yuumasato) [reply linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_group/rule.yml]: I should be correct now.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_gshadow/rule.yml [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+ocil_clause: '{{{ ocil_clause_file_group_owner(file="/etc/gshadow-", group="root") }}}'
+
+ocil: '{{{ ocil_file_group_owner(file="/etc/gshadow-", group="root") }}}'
+
+template:
+    name: file_groupowner
+    vars:
+        filepath: /etc/gshadow-
+        filegid: '0'
+        filegid@debian8: '42'
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_groupowner_backup_etc_gshadow/rule.yml]: I believe that if you make customizations per  product, this should be also reflected in description / ocil.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+ocil_clause: '{{{ ocil_clause_file_group_owner(file="/etc/shadow-", group="root") }}}'
+
+ocil: '{{{ ocil_file_group_owner(file="/etc/shadow-", group="root") }}}'
+
+template:
+    name: file_groupowner
+    vars:
+        filepath: /etc/shadow-
+        filegid: '0'
+        filegid@debian8: '42'
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml]: I believe that if you make customizations per product, this should be also reflected in description / ocil.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_gshadow/rule.yml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+ocil: |-
+    {{{ ocil_file_permissions(file="/etc/gshadow-", perms="----------") }}}
+
+template:
+    name: file_permissions
+    vars:
+        filepath: /etc/gshadow-
+        filemode: '0000'
+        filemode@debian8: '0640'
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_gshadow/rule.yml]: I believe that if you make customizations per product, this should be also reflected in description / ocil.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_shadow/rule.yml [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+ocil: |-
+    {{{ ocil_file_permissions(file="/etc/shadow-", perms="----------") }}}
+
+template:
+    name: file_permissions
+    vars:
+        filepath: /etc/shadow-
+        filemode: '0000'
+        filemode@debian8: '0640'
```
Comment (vojtapolasek) [inline linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_shadow/rule.yml]: I believe that if you make customizations per product, this should be also reflected in description / ocil.

File: linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml (hunk 14 above)
Comment (yuumasato) [reply linux_os/guide/system/permissions/files/permissions_important_account_files/file_owner_backup_etc_shadow/rule.yml]: The text of rules `file_owner_backup_etc_shadow` and `file_groupowner_backup_etc_shadow` were swapped. Fixed by rebase.

**Comment by yuumasato** [comment]
The group and mode are should now be correctly displayed in rule description. I have piggybacked the same changes for the other shadow related rules.

**Comment by yuumasato** [comment]
>  Also you add rules only to rhel8 profile, not to the rhel7. Could you add them as well?

At the time of this PR was proposed, the rhel7 CIS profile didn't exist yet, moreover they were selected in #5333 .
Now, I don't know how that PR finished CI successfully, because the rule doesn't exist there.

**Comment by vojtapolasek** [comment]
I am confused, you linked this PR
https://github.com/ComplianceAsCode/content/pull/5333
But this PR changes only CPE, no modifications to the rhel7 CIS profile.

**Comment by yuumasato** [comment]
@vojtapolasek Sorry, it was supposed to #5331

**Comment by vojtapolasek** [comment]
Ok, seems fine in that case, approving. Thank you.

**Pull request merged**